### PR TITLE
allow users to create their own wrappers

### LINKED ISSUE #16718: Doesn't work with Materialize UI
The materialize input box stop showing its default animation when clicked on a input box although autocomplete works.

``` html
<input id="demo" type="text" class="validate tooltipped awesomplete" data-position="bottom" data-delay="10" data-tooltip="Demo">
```

``` javascript
    var demo_input = document.getElementById("demo"); 
    new Awesomplete(demo_input, {
        list: ["W141001900", "W141001901", "W141001903", "W131001229" "W111001089"],
        minChars: 3,
        maxItems: 20
    });
```

### PR DESCRIPTION
**Pull request by erquhart**
This fixes #16718. Some layouts require the targeted input to remain siblings with other elements - this PR accommodates through the following changes:
- allow the user to create their own wrapper div via the `awesomplete-wrapper` class
- if an input bearing the `awesomplete` class does not have such a wrapper at runtime, Awesomplete will make one just as before, albeit with the `awesomplete-wrapper` class for consistency/simplicity
- CSS selectors updated accordingly
- docs updated as well (hopefully they work, you'll want to test)

My apologies for not writing tests for this, didn't have time to get to it.

### CODE REVIEW

File: awesomplete.js [hunk 1]
```diff
@@ -39,10 +39,12 @@ var _ = function (input, o) {
 
 	// Create necessary elements
 
-	this.container = $.create("div", {
-		className: "awesomplete",
-		around: input
-	});
+	this.container = input.parentNode.className.search(/awesomplete-wrapper/) >= 0 ?
```
Comment (LeaVerou) [inline awesomplete.js:42]: why not `classList.contains("awesomplete-wrapper")`?
(if the answer is "browser support", IE9 can go fu…erm use a polyfill)

File: awesomplete.js [hunk 2]
```diff
@@ -39,10 +39,12 @@ var _ = function (input, o) {
...
+		input.parentNode :
```
Comment (LeaVerou) [inline awesomplete.js:43]: No need for a ternary here. You can just do something like `$(".awesomplete-wrapper", parent) || $.create(...)`

File: index.html [hunk 3]
```diff
@@ -47,6 +47,15 @@
 <p>Before you try anything, you need to include <code>awesomplete.css</code> and <code>awesomplete.js</code> in your page, via the usual <code>&lt;link rel="stylesheet" href="awesomplete.css" /></code> and <code>&lt;script src="awesomplete.js" async>&lt;/script></code> tags.</p>
 
 <p>For the autocomplete, you just need an <code>&lt;input></code> text field (might work on <code>&lt;textarea></code> and elements with <code>contentEditable</code>, but that hasn’t been tested). Add <code>class="awesomplete"</code> for it to be <strong>automatically processed</strong> (you can still specify many options via HTML attributes), otherwise you can instantiate with a few lines of JS code, which allow for more customization.</p>
+
+<p><strong>Note:</strong> Awesomplete will wrap your input in a div with the class 'awesomplete-wrapper'. If other elements need to stay with your input, such as a label, you can simply make the wrapper yourself and Awesomplete will pick it up:
+
+<pre class="language-markup"><code>&lt;div class="awesomplete-wrapper">
```
Comment (LeaVerou) [inline index.html:53]: It would be good to show an actual label here, not another div.

**Comment by LeaVerou** [comment]
Good idea, thanks! See comments. Sorry for the delay in replying.

File: awesomplete.js (hunk 1 above)
Comment (erquhart) [inline awesomplete.js:42]: Yep, that's why. Carries no real cost, so might as well.

File: awesomplete.js (hunk 2 above)
Comment (meganatas) [inline awesomplete.js:43]: help

**Comment by TxHawks** [comment]
Has this been abandoned? I'd be happy to incorporate Lea's comments into this.

Should I just open a new pull request as there seem to be conflicts with current code?

**Comment by AlexDaniel** [comment]
@TxHawks you can fork  @erquhart's repo and do your work there, then submit a pull request here.

**Comment by LeaVerou** [comment]
Yeah, this PR seems to be abandoned. :(
Closing, please send a new one.

**Comment by erquhart** [comment]
Is there a conflict or just these style pref comments?

**Comment by LeaVerou** [comment]
Just these style pref comments :)

**Comment by erquhart** [comment]
Cool. I moved to a different solution, but you're welcome to use the feature addition if someone wants to make the adaptations.

**Pull request closed without merging**